test.js

```diff
@@ -21,3 +21,25 @@ test('emojis over text', function (t) {
   t.is(e.toEmoji('heart'), '❤️')
   t.is(e.toShortCode('❤️'), 'heart')
 })
+
+test('custom shortcodes to emojis', function (t) {
+  t.is(e.toEmoji('keet_laughs'), '😂')
+  t.is(e.toEmoji('keet_party'), '🥳')
+  t.is(e.toEmoji('keet_love'), '😍')
+  t.is(e.toEmoji('keet_music'), '😌🎶')
+  t.is(e.toEmoji('keet_bitcoin'), 'bitcoin')
+  t.is(e.toEmoji('keet_tether'), 'tether')
+  t.is(e.toEmoji('keet_pear'), '🍐')
+  t.is(e.toEmoji('holepunch'), '🕳️🥊')
+
+  t.is(e.toShortCode('😂'), 'joy')
+  t.is(e.toShortCode('🥳'), 'keet_party') // should be 'hooray'
+  t.is(e.toShortCode('😍'), 'keet_love') // should be 'heart_eyes'
+  t.is(e.toShortCode('😌'), 'relieved')
+  t.is(e.toShortCode('🎶'), 'musical_notes')
+  t.is(e.toShortCode('bitcoin'), 'keet_bitcoin')
```

Comment:
I see there is an option to search not only by "short-code": `pray` and `people_hugging` examples found for the `thank` search

Reply:
I would expect to have a one2one short-code --- emoji relationship, can't actually get the reason to have a single emoji for multiple shortcodes, otherwise we need to improve the search? to provide "keet_bitcoin" emoji for any "bit" or "coin" string search?

Reply:
I agree with @mialso here, no need for prefixing

Reply:
@mialso I mean that short code should be uniq. It is not about search. Emoji shortcode is a uniq key. For search we are using tags.
```
  {
    "label":"Keet Music",
    "tags":["keet", "music"], // < -- tags, not uniq.
    "shortCodes": ["keet_music"], // <-- shortcodes - should be uniq
    "emoji":"keet_music",
    "alt": "😌🎶",
    "path": "keet_music.png",
    "pathRiv": "keet_music.riv"
  },
```
When you type `thank` slack is searching 'thank' string in tags and also in shortcodes. Even shortcode not includes `thank` string `pray` icon will be shown because `pray` emoji includes`thanks` in tags as I know.

If one day 'bitcoin' emoji will be included to base emojis set then our solution will be broken because we will have 2 emojis with the same 'bitcoin' shortcode. @holepunchto/emojis will throw exception. How should code know what icon it needs  render if 2 emojis have the same shortcode 'bitcoin' and text includes `:bitcoin:` ? If message includes a emoji as character - then no problems. But If message includes :bitcoin: shortcode then we will have a conflict. I mean, now I can addd just a 'bitcoin' icon because it is a uniq short code.

Reply:
I agree, just wanna say the "search" and "tag(key) uniqueness" are separate problems - I mean neither "bitcoin" nor "keet_bitcoin" solves uniqueness problem completely - still need some logic, taking cases if user is allowed to create own tags and upload custom emojis as well